### PR TITLE
Documentation on docstrings

### PR DESCRIPTION
**Pull request by davidcok**
Accompanies PR 3756

<small>By submitting this pull request, I confirm that my contribution is made under the terms of the [MIT license](https://github.com/dafny-lang/dafny/blob/master/LICENSE.txt).</small>

### CODE REVIEW

File: docs/DafnyRef/Grammar.md [hunk 1]
```diff
@@ -173,6 +173,99 @@ method m() {
...
+* Doc-comments after the declaration are preferred.
+* If the first of a series of single-line or multi-line comments is interpreted as a doc-string, then any subsequent comments
+  are appended to it, so long as there are no intervening lines, whether blank, all white-space or containing program text.
+* Within a multi-line doc-comment, on any line after the first one, 
+    * initial white-space is removed from the string if followed by a '*/'
+    * if there is initial white-space followed by a '*' or '* ', the white-space and
+      star and the optional space or tab character are removed
+    * the opening `/**` (plus any immediately following `*` characters and the closing
+      `*/` along with any immediately preceding '*' characters are also not part of the
+      documentation string
```
Comment (MikaelMayer) [inline docs/DafnyRef/Grammar.md:201]:
```suggestion
* The extraction of the doc-string from a multiline comment follow these rules 
  * On the first line, an optional `*` right after `/*`and an optional space are removed, if present
  * On other lines, the indentation space (with possibly one star in it) is removed, as if the content was supposed to align with A if the comment started with `/** A` for example.
```

**Review by MikaelMayer — CHANGES_REQUESTED** [review]
Tiny changes

File: docs/DafnyRef/Grammar.md [hunk 2]
```diff
@@ -173,6 +173,99 @@ method m() {
...
 comment, leaving trailing text that will provoke parsing errors.
 
+## 2.5. Documentation comments
+
+Like many other languages, Dafny permits _documentation comments_ in a program file.
+Such comments contain natural language descriptions of program elements and may be used
+by IDEs and documentation generation tools to present information to users.
+
+In Dafny programs.
+* Documentation comments are recommended to begin with `/**`, though some
```
[the remark this answers is not in the document]
Comment (davidcok) [reply docs/DafnyRef/Grammar.md:183]: Rephrased

File: docs/DafnyRef/Grammar.md [hunk 3]
```diff
@@ -173,6 +173,99 @@ method m() {
...
+* Documentation comments are recommended to begin with `/**`, though some 
+  comments that begin with `//` or /*` are also interpreted as doc-comments
+* Doc-comments may be associated with any declaration, including type definitions and export declarations.
+* They may be placed before or after the declaration. 
+   * If before, it must be a `/**` comment and may not have any blank or white-space lines between the comment
+     and the declaration.
+   * If after any comments are placed after the signature (with no intervening lines), but before any
+     specifications or left-brace that starts a body, and may be `//` or `/**` comments.
+   * If doc-comments are in both places the content is concatenated.
+* Doc-comments after the declaration are preferred.
```
[the remark this answers is not in the document]
Comment (davidcok) [reply docs/DafnyRef/Grammar.md:192]: OK

**Review by MikaelMayer — APPROVED** [review]

**Pull request merged**